gqcp/tests/ONVBasis/OperatorString_test.cpp

```diff
@@ -0,0 +1,177 @@
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#define BOOST_TEST_MODULE "OperatorString"
+
+#include <boost/test/unit_test.hpp>
+
+#include "ONVBasis/SpinResolvedOperatorString.hpp"
+#include "ONVBasis/SpinUnresolvedOperatorString.hpp"
+#include "QuantumChemical/Spin.hpp"
+
+
+/**
+ *  Check whether the SpinUnresolvedOperatorString constructor works.
+ */
+BOOST_AUTO_TEST_CASE(test_unresolved_constructor) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    std::vector<size_t> indices = {1, 4, 7, 8};
+
+    // Check the constructor.
+    BOOST_CHECK_NO_THROW(GQCP::SpinUnresolvedOperatorString operator_string {indices});
+}
+
+/**
+ *  Check whether the SpinResolvedOperatorString constructor works.
+ */
+BOOST_AUTO_TEST_CASE(test_resolved_constructor) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    std::vector<size_t> indices = {1, 4, 7, 8};
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    // However, the dimension of this index vector doesn't match the spin vector dimension and should result in an exception being thrown.
+    std::vector<size_t> indices_wrong = {4, 7, 8};
+
+    // Initialize the vector containing the spin components we want to use for our operator string indices.
+    std::vector<GQCP::Spin> spins = {GQCP::Spin::alpha, GQCP::Spin::beta, GQCP::Spin::alpha, GQCP::Spin::beta};
+
+    // Initialize the vector containing the spin components we want to use for our operator string indices.
+    // However, the dimension of this spin vector doesn't match the index vector dimension and should result in an exception being thrown.
+    std::vector<GQCP::Spin> spins_wrong = {GQCP::Spin::alpha, GQCP::Spin::beta, GQCP::Spin::alpha};
+
+    // Check the constructor.
+    BOOST_CHECK_THROW(GQCP::SpinResolvedOperatorString operator_string_wrong(indices_wrong, spins), std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::SpinResolvedOperatorString operator_string_wrong_2(indices, spins_wrong), std::invalid_argument);
+    BOOST_CHECK_NO_THROW(GQCP::SpinResolvedOperatorString operator_string_correct(indices, spins));
+}
+
+/**
+ *  Check whether the SpinUnresolvedOperatorString stores its information correctly.
+ */
+BOOST_AUTO_TEST_CASE(test_unresolved_indices) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    std::vector<size_t> indices = {1, 4, 7, 8};
+
+    // Initialize the SpinUnresolvedOperatorString.
+    const auto operator_string = GQCP::SpinUnresolvedOperatorString {indices};
```

Comment:
```suggestion
    const GQCP::SpinUnresolvedOperatorString operator_string {indices};
```
In your statement, you're doing a construction and then an assignment. In this statement, you'll only be doing the construction. C++ isn't Python!